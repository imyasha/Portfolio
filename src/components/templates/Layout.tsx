import React from 'react';
import styled, { ThemeProvider, createGlobalStyle } from 'styled-components';
import { motion, AnimatePresence } from 'framer-motion';
import theme from 'styled-theming';

import useTheme from 'Hooks/useTheme';
import Header from 'Organisms/Header';
import Footer from 'Organisms/Footer';

export const getBackground = theme('mode', {
  light: '#d6deeb',
  dark: '#011627',
});

export const getTextColor = theme('mode', {
  light: '#011627',
  dark: '#d6deeb',
});

const duration = 0.5;

const variants = {
  initial: {
    opacity: 0,
  },
  enter: {
    opacity: 1,
    // filter: 'blur(0)',
    transition: {
      duration,
      delay: duration,
      when: 'beforeChildren',
    },
  },
  exit: {
    opacity: 0,
    // filter: 'blur(1rem)',
    transition: { duration },
  },
};

const Style = createGlobalStyle`
  body {
      background-color: ${getBackground};
      color: ${getTextColor};
      font-family: "brandon-grotesque", sans-serif;
      font-size: 1.25rem;
      margin: 0;
      transition: 0.5s;
  }

  a {
    color: inherit;
    text-decoration: none;
    transition: 0.5s;
  }

  .menu-background {
    background: ${getTextColor};
  }
`;

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  min-height: 95vh;
`;

const Main = styled(motion.main).attrs(() => ({
  initial: 'initial',
  variants,
}))`
  /* flex-grow: 1; */
`;

interface IProps {
  children: React.ReactNode;
  location: any;
}

const Layout = ({ children, location }: IProps) => {
  const colourTheme = useTheme();

  return (
    <ThemeProvider theme={colourTheme}>
      <>
        <Style />
        <Wrapper>
          <Header />
          <AnimatePresence>
            <Main
              key={location?.pathname}
              variants={variants}
              initial="initial"
              animate="enter"
              exit="exit"
            >
              {children}
            </Main>
          </AnimatePresence>
          <Footer />
        </Wrapper>
      </>
    </ThemeProvider>
  );
};

export default Layout;
