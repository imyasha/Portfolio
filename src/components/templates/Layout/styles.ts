import styled from 'styled-components';

const StyledLayout = styled.div`
	min-height: 100vh;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
`;

export { StyledLayout };
