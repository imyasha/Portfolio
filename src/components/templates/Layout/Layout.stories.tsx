import { Layout } from './Layout';

export default {
	title: 'templates/Layout',
	component: Layout,
};

export const LayoutStory = () => <Layout />;

LayoutStory.storyName = 'Layout';
LayoutStory.args = {};
