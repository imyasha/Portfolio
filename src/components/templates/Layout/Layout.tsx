import { FC } from 'react';
import { Footer } from 'Organisms/Footer';
import { Header } from 'Organisms/Header';
import { StyledLayout } from './styles';

const Layout: FC = ({ children }) => {
	return (
		<StyledLayout>
			<Header />
			<main>{children}</main>
			<Footer />
		</StyledLayout>
	);
};

export { Layout };
