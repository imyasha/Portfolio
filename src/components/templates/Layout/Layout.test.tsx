import { Layout } from '../Layout';
import { render } from '@testing-library/react';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('Layout', () => {
	it('renders correctly', () => {
		const { container } = render(<Layout />);
		expect(container).toMatchSnapshot();
	});
});
