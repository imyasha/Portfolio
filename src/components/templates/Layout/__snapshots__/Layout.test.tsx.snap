// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    class="styles__StyledLayout-sc-14vqbs-0 kknGNE"
  >
    <header>
      <div
        class="styles__StyledContainer-mepagl-0 exxZos"
      >
        <div
          class="styles__Flex-sc-14e6kmf-0 iaPxwS"
        >
          <h1>
            <a
              href="/"
            >
              Jacob Herper
            </a>
          </h1>
          <div
            class="styles__Desktop-sc-14e6kmf-1 grzcZM"
          >
            <nav
              class="styles__StyledNavigation-sc-16epmp-0"
            >
              <ul
                class="styles__List-sc-16epmp-1 gPSCMv"
              >
                <li>
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li>
                  <a
                    href="/about"
                  >
                    About
                  </a>
                </li>
                <li>
                  <a
                    href="/uses"
                  >
                    Uses
                  </a>
                </li>
                <li>
                  <a
                    href="/blog"
                  >
                    Blog
                  </a>
                </li>
                <li>
                  <a
                    href="/portfolio"
                  >
                    Portfolio
                  </a>
                </li>
                <li
                  class="desktop-only"
                >
                  <button
                    class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
                    title="Switch to light theme"
                  >
                    <test-file-stub />
                  </button>
                </li>
              </ul>
            </nav>
          </div>
          <div
            class="styles__Mobile-sc-14e6kmf-2 sgtTF"
          >
            <ul
              class="styles__StyledMobileNavigation-sc-14uzdzl-0 crjHjf"
            >
              <li>
                <button
                  class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
                  title="Switch to light theme"
                >
                  <test-file-stub />
                </button>
              </li>
              <li>
                <button
                  class="styles__Menu-sc-14uzdzl-1 dJJTrF"
                >
                  <test-file-stub />
                </button>
              </li>
            </ul>
            <div
              class="styles__Overlay-sc-14uzdzl-2 grizif"
            >
              <nav
                class="styles__StyledNavigation-sc-16epmp-0"
              >
                <ul
                  class="styles__List-sc-16epmp-1 gPSCMv"
                >
                  <li>
                    <a
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li>
                    <a
                      href="/about"
                    >
                      About
                    </a>
                  </li>
                  <li>
                    <a
                      href="/uses"
                    >
                      Uses
                    </a>
                  </li>
                  <li>
                    <a
                      href="/blog"
                    >
                      Blog
                    </a>
                  </li>
                  <li>
                    <a
                      href="/portfolio"
                    >
                      Portfolio
                    </a>
                  </li>
                  <li
                    class="desktop-only"
                  >
                    <button
                      class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
                      title="Switch to light theme"
                    >
                      <test-file-stub />
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main />
    <footer
      class="styles__StyledFooter-razpbj-0 jZveNS"
    >
      <div
        class="styles__StyledContainer-mepagl-0 exxZos"
      >
        <div
          class="styles__Flex-razpbj-1 gJdgaN"
        >
          <span>
            Â© 
            2021
             Jacob Herper
          </span>
          <ul
            class="styles__StyledSocialIcons-sc-15vq785-0 iHrrTj"
          >
            <li>
              <a
                href="https://github.com/jakeherp"
                rel="noopener noreferrer"
                target="_blank"
                title="Github"
              >
                <test-file-stub />
              </a>
            </li>
            <li>
              <a
                href="https://twitter.com/jakeherp"
                rel="noopener noreferrer"
                target="_blank"
                title="Twitter"
              >
                <test-file-stub />
              </a>
            </li>
            <li>
              <a
                href="https://www.linkedin.com/in/jacobherper"
                rel="noopener noreferrer"
                target="_blank"
                title="LinkedIn"
              >
                <test-file-stub />
              </a>
            </li>
            <li>
              <a
                href="https://instagram.com/jakeherp"
                rel="noopener noreferrer"
                target="_blank"
                title="Instagram"
              >
                <test-file-stub />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
